### PR TITLE
feat: add plan upgrading modal for git sync

### PR DESCRIPTION
**Pull request by ihexxa**
Changes:
- add plan upgrading modal for git sync
- add `hideYes` and `hideNo` to `AskModal` to hide buttons

### CODE REVIEW

File: packages/insomnia/src/ui/components/dropdowns/sync-dropdown.tsx [hunk 1]
```diff
@@ -347,6 +356,28 @@ export const SyncDropdown: FC<Props> = ({ vcs, workspace, project }) => {
...
 
+  const gitSyncIsEnabled: FeatureMetadata = JSON.parse(currentOrg?.metadata?.canGitSync || featureMetadataDefaultValue);
+  const accountId = getAccountId();
+  const showUpgradePlanModal = () => {
+    showModal(AskModal, {
+      title: 'Upgrading Plan',
+      message:
+        accountId === currentOrg?.metadata?.ownerAccountId ?
+          'Git Sync feature is only enabled for Team plan or above, please upgrade your plan to continue using this feature.' :
+          'Git Sync feature is only enabled for Team plan or above, please ask organization owner to upgrade plan and continue using this feature.'
```
Comment (gatzjames) [inline packages/insomnia/src/ui/components/dropdowns/sync-dropdown.tsx]:
```suggestion
          'Git Sync feature is only enabled for Team plan or above, please ask the organization owner to upgrade your plan to continue using this feature.'
```

File: packages/insomnia/src/models/organization.ts [hunk 2]
```diff
@@ -2,9 +2,18 @@ interface Branding {
   logo_url: string;
 }
 
+export const featureMetadataDefaultValue = '{ "enabled": false, "reason": "original feature value is undefined" }';
```
Comment (gatzjames) [inline packages/insomnia/src/models/organization.ts]: Do we still need the reason? Since we show custom description about why the feature is not enabled

**Review by gatzjames — APPROVED** [review]
We need to first at least send JSON instead of a string from the API

File: packages/insomnia/src/ui/components/dropdowns/sync-dropdown.tsx [hunk 3]
```diff
@@ -347,6 +357,44 @@ export const SyncDropdown: FC<Props> = ({ vcs, workspace, project }) => {
...
+        yesText: 'Upgrade',
+        noText: 'Cancel',
+        onDone: async (isYes: boolean) => {
+          if (isYes) {
+            window.main.openInBrowser(`${getAppWebsiteBaseURL()}/app/subscription/update?plan=team`);
+          }
+        },
+      }) : showModal(AlertModal, {
+        title: 'Upgrade Plan',
+        message: 'Git Sync is only enabled for Team plan or above, please ask the organization owner to upgrade.',
```
Comment (jackkav) [inline packages/insomnia/src/ui/components/dropdowns/sync-dropdown.tsx:394]: ideally this message should come from the backend "reason": field

**Review by filfreire — APPROVED** [review]

**Comment by ihexxa** [comment]
Let me merge this at first and at least make it work.

**Pull request merged**